packages/babel-parser/src/plugins/v8intrinsic.js

```diff
@@ -0,0 +1,48 @@
+import type Parser from "../parser";
+import { types as tt } from "../tokenizer/types";
+import * as N from "../types";
+
+export default (superClass: Class<Parser>): Class<Parser> =>
+  class extends superClass {
+    markV8IntrinsicStart(): void {
+      if (this.match(tt.modulo)) {
+        this.state.V8IntrinsicStart = this.state.start;
+        this.eat(tt.modulo);
+      }
+    }
+
+    /* ============================================================ *
+     * parser/expression.js                                         *
+     * ============================================================ */
+
+    createIdentifier(): N.Identifier {
+      const identifier = super.createIdentifier(...arguments);
+      if (this.state.V8IntrinsicStart !== undefined) {
```

Comment:
Are there any way to document the type of `V8IntrinsicStart` be a `Number` or `Undefined` ?

Reply:
You should specify it at https://github.com/babel/babel/blob/master/packages/babel-parser/src/tokenizer/state.js

Nit: I know that it's name is V8, but could you start this variable name with `v8`? :sweat_smile: We only use upper-first names for classes/interfaces.

Reply:
Good to know the naming convention here. Resolved as we have removed `V8IntrinsicStart` internal state.